python_modules/dagster/dagster/_core/definitions/asset_key.py

```diff
@@ -0,0 +1,189 @@
+import re
```

Comment:
directly copies over `AssetKey` code from the other file to avoid circular imports